src/vm/object_model.rs

```diff
@@ -228,3 +219,54 @@ pub trait ObjectModel<VM: VMBinding> {
     /// * `object`: The object to be dumped.
     fn dump_object(object: ObjectReference);
 }
+
+pub mod specs {
+    use crate::util::constants::LOG_BITS_IN_WORD;
+    use crate::util::metadata::{
+        header_metadata::HeaderMetadataSpec, side_metadata::SideMetadataSpec, MetadataSpec,
+    };
+
+    macro_rules! define_vm_metadata_spec {
+        ($spec_name: ident, $log_num_bits: expr, $is_global: expr) => {
+            pub struct $spec_name(MetadataSpec);
+            impl $spec_name {
+                const LOG_NUM_BITS: usize = $log_num_bits;
+                const IS_GLOBAL: bool = $is_global;
+                pub const fn in_header(bit_offset: isize) -> Self {
+                    Self(MetadataSpec::InHeader(HeaderMetadataSpec {
+                        bit_offset,
+                        num_of_bits: 1 << Self::LOG_NUM_BITS,
+                    }))
+                }
+                pub const fn side(offset: usize, log_min_obj_size: usize) -> Self {
+                    Self(MetadataSpec::OnSide(SideMetadataSpec {
+                        is_global: Self::IS_GLOBAL,
+                        offset,
+                        log_num_of_bits: Self::LOG_NUM_BITS,
+                        log_min_obj_size,
```

Comment:
`log_min_obj_size` is always `LOG_BYTES_IN_WORD` for all vm_metadata, right?

Reply:
Sorry, that's not true for LOS

Reply:
It seems the only info we need from a binding for side metadata specs is the offset. Everything else can be defined in mmtk-core for each spec.